docs/examples/patterns/state-machine.md

```diff
@@ -0,0 +1,60 @@
+## State Machine
+
+A [state machine](https://en.wikipedia.org/wiki/Finite-state_machine) is simply
+system that exists in one of a number of states. That system changes states, it
+"transitions", due to some input into the system, resulting in another state.
+
+```prolog showLineNumbers
+initialState(0).
+input("add", 1).
+input("subtract", 3).
+input("multiply", 4).
```

Comment:
this runs forever -- i think my understanding about how this works is wrong lol

Reply:
The new value is always different from the previously generated ones so the termination is not guaranteed unless we add a termination condition such as Value < or = something